pkg/config/plain.go

```diff
@@ -67,3 +72,66 @@ func transferOptions(target *model.RuleOptions, source *PlainRuleOptions) {
 		resV.FieldByName(resFieldName).Set(f.Elem())
 	}
 }
+
+func (pcfg *PlainConfig) Validate() error {
+	var errs error
+
+	if pcfg.Default != nil && !slices.Contains(model.DefaultSetValues, model.DefaultSet(*pcfg.Default)) {
+		errs = errors.Join(errs, fmt.Errorf("invalid default set %q; must be one of %q", *pcfg.Default, model.DefaultSetValues))
+	}
+
+	if invalids := getInvalidRules(pcfg.Enable); len(invalids) > 0 {
+		errs = errors.Join(errs, fmt.Errorf("invalid rule name(s) to enable: %q", invalids))
+	}
+
+	if invalids := getInvalidRules(pcfg.Disable); len(invalids) > 0 {
+		errs = errors.Join(errs, fmt.Errorf("invalid rule name(s) to disable: %q", invalids))
+	}
+
+	if len(pcfg.Enable) > 0 && len(pcfg.Disable) > 0 {
+		commons := []string{}
+		for _, e := range pcfg.Enable {
+			for _, d := range pcfg.Disable {
+				if e == d {
+					commons = append(commons, e)
+				}
+			}
+		}
+		if len(commons) > 0 {
+			errs = errors.Join(errs, fmt.Errorf("rule name(s) are both enabled and disabled: %q", commons))
+		}
+	}
+
+	if invalids := getInvalidRegexps(pcfg.Include); len(invalids) > 0 {
+		errs = errors.Join(errs, fmt.Errorf("invalid inclusion pattern(s): %q", invalids))
+	}
+
+	if invalids := getInvalidRegexps(pcfg.Exclude); len(invalids) > 0 {
+		errs = errors.Join(errs, fmt.Errorf("invalid exclusion pattern(s): %q", invalids))
+	}
+
+	if errs != nil {
+		return errs
+	}
+	return nil
```

Comment:
For such use cases, I tend to use a errs []error and uses `append(err, fmt.Errorf...) `

Then use


```suggestion
	if len(errs) > 0 {
		return errors.Join(errs...)
	}
	return nil
```


using consecutive errors.Join on the same errors lead to a lot of error containing error containing errors.... While errors.Join(errs...) won't.